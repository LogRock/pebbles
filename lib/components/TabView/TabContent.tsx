import React, { FC } from "react";

import styled from "styled-components";

import { TabViewProps } from "./TabView.types";

export const TabContentWrapper = styled.div<Pick<TabViewProps, "bgColor">>`
  min-height: 100px;

  font-family: ${({ theme }) => theme.tabView.fontFamily};
  font-size: ${({ theme }) => theme.tabView.fontSize};
  font-weight: ${({ theme }) => theme.tabView.fontWeight};
`;

const TabContent: FC<TabViewProps> = ({ bgColor = "#fff", children }) => {
  return <TabContentWrapper bgColor={bgColor}>{children}</TabContentWrapper>;
};

export default TabContent;
