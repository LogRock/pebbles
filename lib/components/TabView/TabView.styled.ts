import styled from "styled-components";

export const TabViewWrapper = styled.div<{ variant: string }>`
  color: ${(props: any) => props.theme.colors[props.variant || "primary"][500]};
`;
