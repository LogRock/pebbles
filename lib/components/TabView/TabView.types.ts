export interface TabViewProps {
  isActive?: boolean;
  disabled?: boolean;
  bgColor?: string;
  borderRadius?: number;
}
