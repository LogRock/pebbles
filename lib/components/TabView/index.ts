export { default } from "./TabView";
export type { TabViewProps } from "./TabView.types";
