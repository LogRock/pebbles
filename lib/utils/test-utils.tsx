/* eslint-disable import/export */
import React from "react";
import { cleanup, render } from "@testing-library/react";
import { afterEach } from "vitest";
import { ThemeProvider } from "styled-components";
import { light as theme } from "../theme";

afterEach(() => {
  cleanup();
});

const customRender = (ui: React.ReactElement, options = {}) =>
  render(ui, {
    // wrap provider(s) here if needed
    wrapper: ({ children }) => (
      <ThemeProvider theme={theme}>{children}</ThemeProvider>
    ),
    ...options,
  });

export * from "@testing-library/react";
export { default as userEvent } from "@testing-library/user-event";
// override render export
export { customRender as render };
