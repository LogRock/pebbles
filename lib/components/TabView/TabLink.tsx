import React, { FC } from "react";
import styled, { css } from "styled-components";

import { TabLinkProps } from "./TabView.types";

const TabLinkAnchor = styled.a<TabLinkProps>`
  display: flex;
  position: relative;
  align-items: flex-end;
  gap: ${({ theme }) => theme.spacings.xsm};

  margin: 0 ${({ theme }) => theme.spacings.md};
  padding-top: 0;
  padding-right: ${({ theme }) => theme.spacings.xbig};
  padding-bottom: ${({ theme }) => theme.spacings.md};
  padding-left: ${({ theme }) => theme.spacings.xbig};

  color: ${({ theme, active }) =>
    active ? theme.colors.neutral[900] : theme.colors.neutral[400]};
  font-family: ${({ theme }) => theme.tabView.fontFamily};
  font-size: ${({ theme }) => theme.tabView.fontSize};
  font-weight: ${({ theme }) => theme.tabView.fontWeight};
  line-height: 1.5;
  text-align: center;

  cursor: ${({ disabled }) => (disabled ? "not-allowed" : "pointer")};

  &:hover {
    color: ${({ theme, disabled }) => !disabled && theme.colors.neutral[900]};
  }

  ${({ active, theme }) => {
    if (active) {
      return css`
        &:after {
          content: " ";

          position: absolute;
          right: 0;
          bottom: 0;
          left: 0;
          width: 100%;
          height: 4px;

          border-top-left-radius: 4px;
          border-top-right-radius: 4px;

          background: ${theme.colors.primary[500]};
        }
      `;
    }
    return "";
  }}
`;

const TabLink: FC<TabLinkProps> = ({ disabled, href, ...props }) => (
  <TabLinkAnchor
    href={disabled ? undefined : href}
    disabled={disabled}
    {...props}
  />
);

export default TabLink;
