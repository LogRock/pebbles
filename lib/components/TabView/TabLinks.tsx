import React, { FC } from "react";

import styled from "styled-components";

import { TabViewProps } from "./TabView.types";

export const TabLinksWrapper = styled.div<
  Pick<TabViewProps, "bgColor" | "justify">
>`
  display: flex;
  align-items: center;
  justify-content: ${({ justify }) => justify};
  padding: ${({ theme }) => `${theme.spacings.md} ${theme.spacings.xbig}`};

  border-radius: 10px 10px 0 0;

  background-color: ${({ bgColor }) => bgColor};
`;

const TabLinks: FC<TabViewProps> = ({
  bgColor = "#fff",
  justify = "center",
  children,
}) => {
  return (
    <TabLinksWrapper bgColor={bgColor} justify={justify}>
      {children}
    </TabLinksWrapper>
  );
};

export default TabLinks;
