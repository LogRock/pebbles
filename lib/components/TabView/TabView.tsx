import React, { FC } from "react";
import { TabViewWrapper } from "./TabView.styled";
import { TabViewProps } from "./TabView.types";

const TabView: FC<TabViewProps> = ({
  bgColor = "#fff",
  borderRadius = 10,
  children,
  ...rest
}) => {
  return (
    <TabViewWrapper bgColor={bgColor} borderRadius={borderRadius} {...rest}>
      {children}
    </TabViewWrapper>
  );
};

export default TabView;
