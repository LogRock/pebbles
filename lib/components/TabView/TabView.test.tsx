import React from "react";

import "@testing-library/jest-dom";
import { render, screen } from "../../utils/test-utils";

import TabView from "./TabView";
import TabLinks from "./TabLinks";
import TabLink from "./TabLink";
import TabContent from "./TabContent";

describe("TabView", () => {
  it("renders TabView properly", () => {
    const { container } = render(
      <TabView>
        <TabLinks>
          <TabLink active>Tab 1</TabLink>
          <TabLink>Tab 2</TabLink>
          <TabLink disabled>Tab 3</TabLink>
        </TabLinks>
        <TabContent>
          <h4>Tab Content here</h4>
        </TabContent>
      </TabView>
    );

    expect(screen.getByText("Tab 1") as HTMLAnchorElement).toBeInTheDocument();
    expect(screen.getByText("Tab Content here")).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
