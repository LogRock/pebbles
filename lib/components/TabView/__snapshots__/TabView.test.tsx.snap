// Vitest Snapshot v1

exports[`TabView > renders TabView properly 1`] = `
<div>
  <div
    class="sc-bdvvtL eukWAO"
  >
    <div
      class="sc-gsDKAQ lfSFVJ"
    >
      <a
        class="sc-dkPtRN htQWIk"
      >
        Tab 1
      </a>
      <a
        class="sc-dkPtRN pjDii"
      >
        Tab 2
      </a>
      <a
        class="sc-dkPtRN dBSncu"
        disabled=""
      >
        Tab 3
      </a>
    </div>
    <div
      class="sc-hKwDye fyhOfe"
    >
      <h4>
        Tab Content here
      </h4>
    </div>
  </div>
</div>
`;
